### ISSUE TITLE
Create new Process object only once in Resource Monitoring Windows

### ISSUE THREAD

**Issue by evgenyfedorov2** [dotnet/extensions#5588]
[GetCurrentProcessMemoryUsage](https://github.com/dotnet/extensions/blob/e18a05582e64603a0f2e2871bcf555e476570102/src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Windows/Interop/ProcessInfo.cs#L42 ) method creates a new Process object on every method call:

```
public ulong GetCurrentProcessMemoryUsage()
{
    using Process process = Process.GetCurrentProcess();
    return (ulong)process.WorkingSet64;
}
```

Instead, it could have created it once and then reusing it for the application lifetime. This should improve performance.

**Comment by haipz** [comment]
This might hit performance issue:
![Image](https://github.com/user-attachments/assets/a87ae347-fac2-4e9b-91d9-8c89b82bc629)

**Comment by makazeu** [comment]
Not sure why GetCurrentProcess is really cpu-heavy

**Comment by evgenyfedorov2** [comment]
Fixed in #5597
